mousestyles/dynamics/__init__.py

```diff
@@ -121,3 +122,135 @@ def create_time_matrix(combined_gap=4, time_gap=1, days_index=137):
     title = pd.DataFrame(days, columns=['strain', 'mouse', 'day'])
     time_matrix = pd.concat([title, matrix], axis=1)
     return(time_matrix)
+
+
+def get_prob_matrix_list(time_df, interval_length=1000):
+    r"""
+    returns a list of probability transition matrices
+    that will be later used to characterize and simu-
+    late the behavior dynamics of different strains of
+    mice. The data used as input is the pandas DataFrame
+    generated by function create_time_matrix with de-
+    fault parameters. The output is a list of numpy
+    arrays, each being a transition matrix characterizing
+    one small time interval. The interval length could
+    be chose.
+
+    Parameters
+    ----------
+    time_df: Pandas.DataFrame
+        a huge data frame containing info on strain, mouse
+        no., mouse day, and different states at chosen time
+        points.
+    interval_length: int or float
+        an integer specifying the desired length of each
+        small time interval
+
+    Returns
+    -------
+    matrix_list: list
+        a list of the mle estimations of the probability tran-
+        sition matrices for each small time interval stored in
+        the format of numpy array.
+
+    Examples
+    --------
+    >>> row_i = np.hstack((np.zeros(13), np.ones(10),
+                            np.ones(10)*2, np.ones(10)*3))
+    >>> time_df_eg = np.vstack((row_i, row_i,row_i))
+    >>> mat_list = get_prob_matrix_list(time_df_eg,
+                                        interval_length=10)
+    >>> mat_list[0]
+    >>> array([[ 1.,  0.,  0.,  0.],
+               [ 0.,  0.,  0.,  0.],
+               [ 0.,  0.,  0.,  0.],
+               [ 0.,  0.,  0.,  0.]])
+    >>> mat_list[1]
+    >>> array([[ 0.,  0.,  0.,  0.],
+               [ 0.,  1.,  0.,  0.],
+               [ 0.,  0.,  0.,  0.],
+               [ 0.,  0.,  0.,  0.]])
+    >>> mat_list[2]
+    >>> array([[ 0.,  0.,  0.,  0.],
+               [ 0.,  0.,  0.,  0.],
+               [ 0.,  0.,  1.,  0.],
+               [ 0.,  0.,  0.,  0.]])
+    >>> mat_list[3]
+    >>> array([[ 0.,  0.,  0.,  0.],
+               [ 0.,  0.,  0.,  0.],
+               [ 0.,  0.,  0.,  0.],
+               [ 0.,  0.,  0.,  1.]])
+
+    """
+    # check all the inputs
+    condition_time_df = (type(time_df) == pd.core.frame.DataFrame)
+    condition_interval_length = (type(interval_length) == int and
+                                 interval_length > 0)
+    if not condition_time_df:
+        raise ValueError("time_df should be pandas DataFrame")
+    if not condition_interval_length:
+        raise ValueError("combined_gap should be positive int")
```

Comment:
Combined_gap is different from interval_length? If so, we need more explanations on it.